src/golang/lib/job/databricks.go

```diff
@@ -0,0 +1,189 @@
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksDataScript, specStr, nil
+
+	} else if spec.Type() == SystemMetricJobType {
+		specStr, err := EncodeSpec(spec, JsonSerializationType)
+		if err != nil {
+			return "", "", errors.Wrap(err, "Unable to encode spec.")
+		}
+		return databricksMetricScript, specStr, nil
+
+	} else {
+		return "", "", errors.New("Unsupported JobType was passed in.")
+	}
+}
+
+func (j *DatabricksJobManager) Config() Config {
+	return j.conf
+}
+
+func (j *DatabricksJobManager) Launch(
+	ctx context.Context,
+	name string,
+	spec Spec,
+) error {
+	log.Infof("Running %s job %s.", spec.Type(), name)
+
+	scriptFile, specStr, err := j.mapJobTypeToFile(spec)
+	if err != nil {
+		return err
+	}
+	storageConfig, err := spec.GetStorageConfig()
+	if err != nil {
+		return errors.Wrap(err, "Spec unexpectedly has no storage config.")
+	}
+	bucket := storageConfig.S3Config.Bucket
+	pythonFilePath := fmt.Sprintf("%s/%s", bucket, scriptFile)
+
+	jobId, err := databricks_lib.CreateJob(ctx, j.databricksClient, name, j.conf.S3InstanceProfileArn, pythonFilePath)
+	if err != nil {
+		return errors.Wrap(err, "Error creating job in Databricks.")
+	}
+	log.Info("SPEC:")
+	log.Info(specStr)
+	runId, err := databricks_lib.RunNow(ctx, j.databricksClient, jobId, specStr)
+	if err != nil {
+		return errors.Wrap(err, "Error runnning job in Databricks.")
+	}
+	j.runMap[name] = runId
+	return nil
+}
+
+func (j *DatabricksJobManager) Poll(ctx context.Context, name string) (shared.ExecutionStatus, error) {
+	runId, ok := j.runMap[name]
+	if !ok {
+		return shared.UnknownExecutionStatus, ErrJobNotExist
+	}
+
+	getRunReq := &jobs.GetRun{
+		RunId: runId,
+	}
+	getRunResp, err := j.databricksClient.Jobs.GetRun(ctx, *getRunReq)
+	if err != nil {
+		return shared.UnknownExecutionStatus, errors.Wrap(err, "Unable to get run from databricks.")
+	}
+
+	status := getRunResp.State.LifeCycleState
+	if status == jobs.RunLifeCycleStatePending ||
+		status == jobs.RunLifeCycleStateRunning ||
+		status == jobs.RunLifeCycleStateTerminating {
+		return shared.RunningExecutionStatus, nil
+	} else if status == jobs.RunLifeCycleStateInternalError {
+		return shared.FailedExecutionStatus, nil
+	} else if status == jobs.RunLifeCycleStateTerminated {
+		getTerminalRunResp, err := j.databricksClient.Jobs.GetRun(ctx, *getRunReq)
+		if err != nil {
+			return shared.UnknownExecutionStatus, errors.Wrap(err, "Unable to get run from databricks.")
+		}
+		resultStatus := getTerminalRunResp.State.ResultState
+		if resultStatus == jobs.RunResultStateSuccess {
+			return shared.SucceededExecutionStatus, nil
+		} else {
+			return shared.FailedExecutionStatus, nil
+		}
+	} else {
+		return shared.UnknownExecutionStatus, ErrAsyncExecution
+	}
+
+}
+
+func (j *DatabricksJobManager) DeployCronJob(
+	ctx context.Context,
+	name string,
+	period string,
+	spec Spec,
+) error {
+	return nil
```

Comment:
Can we change this to return an error that deploying cron jobs on databricks is not supported